src/mirage_kv_unix.ml

```diff
@@ -46,224 +71,298 @@ let split_string delimiter name =
     let idx = try index_from name off delimiter with _ -> len in
     let fst = sub name off (idx - off) in
     let idx' = idx + 1 in
-    if idx' <= len then
-      doit idx' (fst :: acc)
-    else
-      fst :: acc
+    if idx' <= len then doit idx' (fst :: acc) else fst :: acc
   in
   List.rev (doit 0 [])
 
 let rec remove_dots parts outp =
-  match parts, outp with
-  | ".."::r, _::rt -> remove_dots r  rt
-  | ".."::r, []    -> remove_dots r  []
-  | "."::r , rt    -> remove_dots r  rt
-  | r::rs  , rt    -> remove_dots rs (r :: rt)
-  | []     , rt    -> List.rev rt
+  match (parts, outp) with
+  | ".." :: r, _ :: rt -> remove_dots r rt
+  | ".." :: r, [] -> remove_dots r []
+  | "." :: r, rt -> remove_dots r rt
+  | r :: rs, rt -> remove_dots rs (r :: rt)
+  | [], rt -> List.rev rt
```

Comment:
Alternatively, we could just disallow `.` and `..` as names - and return ``Error (`Not_found _)``. I think other mirage-kv implementations don't treat `.` and `..` specially.

Reply:
It would be super useful to have a mirage-kv-test package that check that all the KV store behave similarly